lrlex/src/lib/lexer.rs

```diff
@@ -376,31 +382,53 @@ if 'IF'
         let mut lexer = lexerdef.lexer("a b c");
         let lexemes = lexer.all_lexemes().unwrap();
         assert_eq!(lexemes.len(), 3);
-        assert_eq!(lexer.offset_line_col(lexemes[1].start()), (1, 3));
+        assert_eq!(lexer.line_col(lexemes[1].start()), (1, 3));
         assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "a b c");
 
         let mut lexer = lexerdef.lexer("a b c\n");
         let lexemes = lexer.all_lexemes().unwrap();
         assert_eq!(lexemes.len(), 3);
-        assert_eq!(lexer.offset_line_col(lexemes[1].start()), (1, 3));
+        assert_eq!(lexer.line_col(lexemes[1].start()), (1, 3));
         assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "a b c");
 
         let mut lexer = lexerdef.lexer(" a\nb\n  c d");
         let lexemes = lexer.all_lexemes().unwrap();
         assert_eq!(lexemes.len(), 4);
-        assert_eq!(lexer.offset_line_col(lexemes[0].start()), (1, 2));
-        assert_eq!(lexer.offset_line_col(lexemes[1].start()), (2, 1));
-        assert_eq!(lexer.offset_line_col(lexemes[2].start()), (3, 3));
-        assert_eq!(lexer.offset_line_col(lexemes[3].start()), (3, 5));
+        assert_eq!(lexer.line_col(lexemes[0].start()), (1, 2));
+        assert_eq!(lexer.line_col(lexemes[1].start()), (2, 1));
+        assert_eq!(lexer.line_col(lexemes[2].start()), (3, 3));
+        assert_eq!(lexer.line_col(lexemes[3].start()), (3, 5));
         assert_eq!(lexer.surrounding_line_str(lexemes[0].start()), " a");
         assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "b");
         assert_eq!(lexer.surrounding_line_str(lexemes[2].start()), "  c d");
         assert_eq!(lexer.surrounding_line_str(lexemes[3].start()), "  c d");
     }
 
+    #[test]
+    fn test_line_col_multibyte() {
+        let src = "%%
+[a-z❤]+ 'ID'
+[ \\n] ;"
+            .to_string();
+        let mut lexerdef = parse_lex(&src).unwrap();
+        let mut map = HashMap::new();
+        map.insert("ID", 0u8);
+        assert_eq!(lexerdef.set_rule_ids(&map), (None, None));
+
+        let mut lexer = lexerdef.lexer(" a\n❤ b");
+        let lexemes = lexer.all_lexemes().unwrap();
+        assert_eq!(lexemes.len(), 3);
+        assert_eq!(lexer.line_col(lexemes[0].start()), (1, 2));
+        assert_eq!(lexer.line_col(lexemes[1].start()), (2, 1));
+        assert_eq!(lexer.line_col(lexemes[2].start()), (2, 3));
+        assert_eq!(lexer.surrounding_line_str(lexemes[0].start()), " a");
+        assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "❤ b");
+        assert_eq!(lexer.surrounding_line_str(lexemes[1].start()), "❤ b");
```

Comment:
I think this last assert was meant to be `assert_eq!(lexer.surrounding_line_str(lexemes[2].start()), "b");`

Reply:
Oops, deliberate mistake! [I might be lying.] I'll fix it.